File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
using namespace std;

int main()
{
	int n, x;
	cin >> n;
	stack<int> stk;
	string str;
	
	for (int i = 0; i < n; i++) {
		cin >> str;
		if (str == "push") {
			cin >> x;
			stk.push(x);
		}
		else if (str == "top") {
			if (!stk.empty())
				cout << stk.top() << '\n';
			else
				cout << "-1" << '\n';
		}
		else if (str == "size") {
			cout << stk.size() << '\n';
		}
		else if (str == "empty") {
			if (stk.empty())
				cout << "1" << '\n';
			else
				cout << "0" << '\n';
		}
		else if (str == "pop") {
			if (!stk.empty()) {
				cout << stk.top() << '\n';
				stk.pop();
			}
			else {
				cout << "-1" << '\n';
			}
		}
	}
	return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		Stack<Integer> stack = new Stack<>();
		int n = Integer.parseInt(br.readLine());
		for (int i = 0; i < n; i++) {
			String str = br.readLine();
			String[] arr = str.split(" ");
			
			String order = arr[0];
			if (order.equals("push")) {
				stack.push(Integer.parseInt(arr[1]));
			} else if (order.equals("top")) {
				if (stack.isEmpty()) {
					System.out.println(-1);
				} else {
					System.out.println(stack.peek());
				}
			} else if (order.equals("size")) {
				System.out.println(stack.size());
			} else if (order.equals("empty")) {
				if (stack.isEmpty()) {
					System.out.println(1);
				} else {
					System.out.println(0);
				}
			} else if (order.equals("pop")) {
				if (stack.isEmpty()) {
					System.out.println(-1);
				} else {
					int pop = stack.pop();
					System.out.println(pop);
				}
			}
		}
	}
}
